Confus/Confus/LocalPlayerController.cpp

```diff
@@ -0,0 +1,51 @@
+#include <RakNet/PacketPriority.h>
+
+#include "LocalPlayerController.h"
+#include "../ConfusShared/EventManager.h"
+#include "../ConfusShared/Player.h"
+#include "../ConfusShared/PacketType.h"
+
+namespace Confus
+{
+	LocalPlayerController::LocalPlayerController(ConfusShared::Player& a_Player, Networking::ClientConnection& a_Connection)
+		: m_Player(a_Player),
+		m_Connection(a_Connection)
+	{
+	}
+
+	void LocalPlayerController::handleInput(ConfusShared::EventManager& a_EventManager)
+	{
+		m_InputState.ForwardPressed = m_InputState.ForwardPressed || a_EventManager.IsKeyDown(irr::EKEY_CODE::KEY_KEY_W);
+		m_InputState.BackwardPressed = m_InputState.BackwardPressed || a_EventManager.IsKeyDown(irr::EKEY_CODE::KEY_KEY_S);
+		m_InputState.LeftPressed = m_InputState.LeftPressed || a_EventManager.IsKeyDown(irr::EKEY_CODE::KEY_KEY_A);
+		m_InputState.RightPressed = m_InputState.RightPressed || a_EventManager.IsKeyDown(irr::EKEY_CODE::KEY_KEY_D);
+		if(a_EventManager.IsRightMouseDown())
+		{
+			m_Player.startHeavyAttack();
+		}
+		else if(a_EventManager.IsLeftMouseDown())
+		{
+			m_Player.startLightAttack();
+		};
+	}
+
+	void LocalPlayerController::fixedUpdate()
+	{
+		RakNet::BitStream bitstream;
+		bitstream.Write(static_cast<RakNet::MessageID>(ConfusShared::Networking::EPacketType::Player));
+		bitstream.Write(m_Player.getGUID());
+		bitstream.Write(m_Player.getRotation());
+		bitstream.Write(m_InputState);
+
+		m_Connection.sendMessage(&bitstream, PacketReliability::UNRELIABLE);
+		resetInputState();
+	}
```

Comment:
I agree that it's not the best way of doing things, since fixedupdates may come quickly in succession and are more meant for in game behaviour. However, this was the original approach and I did not want to do much more than rewrite it. We can always look for a better solution afterwards, but I personally don't think it should integrated in this particular pull request in that case